packages/widget-chat/src/actions/flags.js

```diff
@@ -1,50 +1,83 @@
+export const ADD_FLAG_BEGIN = `ADD_FLAG_BEGIN`;
+function addFlagBegin(activity) {
+  return {
+    type: ADD_FLAG_BEGIN,
+    payload: {
+      activity
+    }
+  };
+}
+
 export const ADD_FLAG = `ADD_FLAG`;
-function actionAddFlag(activity) {
+function addFlagSuccess(activity, flag) {
   return {
     type: ADD_FLAG,
-    activity
+    payload: {
+      activity,
+      flag
+    }
   };
 }
 
-export const BEGIN_RECEIVE_FLAGS = `BEGIN_RECEIVE_FLAGS`;
-function actionBeginReceiveFlags() {
+function addFlagError(activity, error) {
   return {
-    type: BEGIN_RECEIVE_FLAGS
+    type: ADD_FLAG,
+    payload: {
+      activity,
+      error
+    },
+    error: true
   };
 }
 
-export const RECEIVE_FLAGS = `RECEIVE_FLAGS`;
-function actionReceiveFlags(flags) {
+export const FLAGS_REQUEST_BEGIN = `FLAGS_REQUEST_BEGIN`;
```

Comment:
Shouldn't this be `REQUEST_FLAGS_BEGIN`? I think we've been naming actions `VERB_NOUN(_VERB)`?